src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1446,6 +1453,15 @@ private void WriteHeaders(HttpRequestMessage request, ref ArrayBuffer headerBuff
                 WriteIndexedHeader(H2StaticTable.PathSlash, pathAndQuery, ref headerBuffer);
             }
 
+            if (request.HasHeaders && request.Headers.Protocol != null)
+            {
+                WriteBytes(":protocol", ref headerBuffer);
+                Encoding? protocolEncoding = _pool.Settings._requestHeaderEncodingSelector?.Invoke(":protocol", request);
+                WriteLiteralHeaderValue(request.Headers.Protocol, protocolEncoding, ref headerBuffer);
+
+                request.Headers.Protocol = null;
```

Comment:
I'm confused how these changes pass tests... setting Protocol to null above throws a FormatException.

Reply:
Agree, I moved it behind the collection